middleware/auth.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+)
+
+func isAuthorized(r *http.Request) bool {
+	// TODO: Add configuration to enable this Auth
+	// This configuration should not be as command line option
+	// Because, one glusterd can be run with `--no-auth` and
+	// other glusterd with auth enabled.
+	if strings.HasPrefix(r.RemoteAddr, "127.0.0.1:") {
+		return true
+	}
+	// TODO: Add Auth for requests from Remote IPs
+	// Without that, this feature is no impact on
+	// existing(Without Auth). Returning true so that it
+	// will not break the existing consumers/tests till Auth introduced
+	return true
+}
+
+// Auth is a middleware which validates the incoming request is from local node or remote node
+func Auth(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if isAuthorized(r) {
+			next.ServeHTTP(w, r)
+		} else {
+			http.Error(w, "Unauthorized", http.StatusUnauthorized)
```

Comment:
A `403` (http.StatusForbidden) is more suited here because as per HTTP [RFC](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) a `401 Unauthorized` response MUST include a WWW-Authenticate header field to allow client to authenticate again.


Also, the body can contain something more specific which says that requests from non-local IPs will not be served.